nanome_rmsd/RMSD.py

```diff
@@ -250,8 +278,68 @@ def align(self, p_complex, q_complex):
             p_cent = p_complex.rotation.rotate_vector(help.array_to_position(p_cent))
             q_cent = q_complex.rotation.rotate_vector(help.array_to_position(q_cent))
             q_complex.position = p_complex.position + p_cent - q_cent
+           
+            if(self._menu.error_message.text_value=="Loading..."):
+                self._menu.change_error("clear")
+            
+            p_complex.locked = True
+            q_complex.locked = True
+        
         return result_rmsd
 
+    # auto select with global/local alignment
+    def select(self,mobile,target):
+        # create seq1 and seq2
+        #run the alignment
+        self._menu.change_error("loading")
+        #self._mobile = mobile
+        self._mobile = mobile
+        # Logs.debug("mobile list before ",self._mobile)
+        self._target = target
+        self.request_workspace(self.on_select_received) 
```

Comment:
If you only want your target and mobile structures, you could write self.request_structures_deep([target] + mobile), then, your callback would receive the target as it's first element, and all the other elements would be the mobiles.
With that, you wouldn't need your first for loop in on_select_received